gpflow/conditionals/util.py

```diff
@@ -196,6 +228,16 @@ def independent_interdomain_conditional(Kmn,
     """
     M, L, N, P = tf.unstack(tf.shape(Kmn), num=Kmn.shape.ndims, axis=0)
 
+    shape_constraints = [
```

Comment:
This function isn't actually tested - probably not scope of this PR to add a test for that, but should we turn it into a separate issue ?

Reply:
not fussed - this PR or another, but yeah somewhere

Reply:
Put into an issue: #1295